IpmiFeaturePkg/IpmiWatchdog/IpmiWatchdog.c

```diff
@@ -0,0 +1,160 @@
+/** @file
+  Common code for handling IPMI Watch Dog Timers.
+
+  Copyright (c) Microsoft Corporation
+  Copyright (c) 2018 - 2019, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/PcdLib.h>
+#include <Library/IpmiCommandLib.h>
+
+#include <IndustryStandard/Ipmi.h>
+
+#include <IpmiWatchdog.h>
+
+/**
+  Enables the IPMI watchdog timer with the provided configurations.
+
+  @param[in]  TimerUse        The current use of the IPMI watchdog timer.
+  @param[in]  TimerAction     The requested action of the BMC if the time expires.
+  @param[in]  FlagsClear      The watchdog expired flags to clear.
+  @param[in]  CountdownValue  The initial time of the watchdog timer in seconds.
+
+  @retval   EFI_SUCCESS         Watchdog timer was successfully started.
+  @retval   EFI_PROTOCOL_ERROR  Unexpected completion code was returned.
+  @retval   Other               An error was returned by the IPMI command lib.
+**/
+EFI_STATUS
+EFIAPI
+IpmiEnableWatchdogTimer (
+  IN CONST UINT8   TimerUse,
+  IN CONST UINT8   TimerAction,
+  IN CONST UINT8   FlagsClear,
+  IN CONST UINT16  CountdownValue
+  )
+{
+  EFI_STATUS                        Status;
+  IPMI_SET_WATCHDOG_TIMER_REQUEST   WatchdogTimer;
+  IPMI_GET_WATCHDOG_TIMER_RESPONSE  GetWatchdogTimer;
+  UINT8                             CompletionCode;
+
+  DEBUG ((
+    DEBUG_INFO,
+    "Setting IPMI watchdog timer. Timer: 0x%x Action: 0x%x ClearFlags: 0x%x Countdown: 0x%x\n",
+    TimerUse,
+    TimerAction,
+    FlagsClear,
+    CountdownValue
+    ));
+
+  ASSERT (TimerUse != 0);
+  ASSERT (CountdownValue != 0);

```

Comment:
In the production FW case, should this function error out in these cases or is it safe for it to continue?

Reply:
Timer use should be statcally defined so this is just sanity, but i'll codify this to be safe.